### PR TITLE
Add try-except when assigning customfield

### PR DESCRIPTION
**Pull request by tom-borcin**
PR fixes an issue that occurs when setting GitHub Issue custom field for a project that doesn't have that custom field enabled.

### CODE REVIEW

**Comment by tom-borcin** [comment]
@tomassebestik please take a look

**Review by mahavirj — APPROVED** [review]

File: sync_issues_to_jira/sync_issue.py [hunk 1]
```diff
@@ -71,8 +71,11 @@ def handle_issue_closed(jira, event):
     # issues often get closed for the wrong reasons - ie the user
     # found a workaround but the root cause still exists.
     issue = _leave_jira_issue_comment(jira, event, "closed", False)
-    # Sets value of custom GitHub Issue field to Closed
-    issue.update(fields={'customfield_12100': {'value': 'Closed'}}) 
+    try :
+        # Sets value of custom GitHub Issue field to Closed
+        issue.update(fields={'customfield_12100': {'value': 'Closed'}})
+    except JIRAError as error:
```
Comment (tomassebestik) [inline sync_issues_to_jira/sync_issue.py:77]: @tom-borcin Are we sure this is a `JIRAError` exception?

It seems to fail with an `AttributeError` error,  https://github.com/espressif/esp-matter/actions/runs/5044110229/jobs/9046736828#step:4:222

Comment (tom-borcin) [reply sync_issues_to_jira/sync_issue.py:77]: I tested it on my own fork and it worked: tom-borcin/github-actions/sync_issues_to_jira@fix/test_custom_field

Linked action fails with: `AttributeError: 'NoneType' object has no attribute 'update'
[222](https://github.com/espressif/esp-matter/actions/runs/5044110229/jobs/9046736828#step:4:223)
}` on this GitHub issue: https://github.com/espressif/esp-matter/issues/396

GH issue wasn't mirrored to Jira and script just line before on [L73](https://github.com/espressif/github-actions/blob/master/sync_issues_to_jira/sync_issue.py#L73) returns None (should return mirrored Jira issue). Calling update on None type causes AttributeError.

Comment (tomassebestik) [reply sync_issues_to_jira/sync_issue.py:77]: @tom-borcin I'm sorry, but I'm a bit confused by this explanation ...

> ..... Calling update on None type causes `AttributeError`.

So in that case we should change it to :
```python
try :
	# Sets value of custom GitHub Issue field to Closed
	issue.update(fields={'customfield_12100': {'value': 'Closed'}})

except AttributeError as error:
	...
```

Or am I missing something? Thank you

@kumekay PTAL as well ... many thanks!

**Review by kumekay — APPROVED** [review]
@tom-borcin Thank you, it should work

**Pull request merged**